Plan/api/src/main/java/com/djrapitops/plan/delivery/web/resolver/request/Request.java

```diff
@@ -34,25 +34,28 @@
     private final URIQuery query;
     private final WebUser user;
     private final Map<String, String> headers;
+    private final URIQuery formBody;
 
     /**
      * Constructor.
      *
-     * @param method  HTTP method, GET, PUT, POST, etc
-     * @param path    Requested path /example/target
-     * @param query   Request parameters ?param=value etc
-     * @param user    Web user doing the request (if authenticated)
-     * @param headers Request headers https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers
+     * @param method   HTTP method, GET, PUT, POST, etc
+     * @param path     Requested path /example/target
+     * @param query    Request parameters ?param=value etc
+     * @param user     Web user doing the request (if authenticated)
+     * @param headers  Request headers https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers
+     * @param formBody Map representation of application/x-www-form-urlencoded POST data, if present
      */
-    public Request(String method, URIPath path, URIQuery query, WebUser user, Map<String, String> headers) {
+    public Request(String method, URIPath path, URIQuery query, WebUser user, Map<String, String> headers, URIQuery formBody) {
```

Comment:
This should not be URIQuery but use bytes or String instead to support different kinds of request bodies. Optional solution would be to put String here as requestBody and then later parse the requestBody when needed. This would support things like json as request body.

```suggestion
    public Request(String method, URIPath path, URIQuery query, WebUser user, Map<String, String> headers, byte[] requestBody) {
```

```suggestion
    public Request(String method, URIPath path, URIQuery query, WebUser user, Map<String, String> headers, String requestBody) {
```

URI query is a concept that represents `?parameter=value&another=1` as a whole https://en.wikipedia.org/wiki/Query_string

Reply:
Okay, I wasn't sure if it needed to be generalized but i had some ideas as to how that could be done